scheduler/drilling-wits-depth-median/tests/test_unit.py

```diff
@@ -0,0 +1,225 @@
+from typing import List
+
+import pytest
+from corva import Api, ScheduledDepthEvent
+from pytest_mock import MockerFixture
+
+from src import api as api_lib
+from src import models, service
+from src.app import app
+
+
+def test_app_returns_early(app_runner, mocker: MockerFixture):
+    """It returns early if no records were fetched from the API."""
+
+    event = ScheduledDepthEvent(
+        asset_id=0,
+        company_id=0,
+        top_depth=0.0,
+        bottom_depth=0.0,
+        log_identifier='',
+        interval=0.0,
+    )
+
+    mocker.patch.object(api_lib, 'get_records', return_value=[])
+    spy = mocker.spy(service, 'group_records_by_milestone')
+
+    app_runner(app, event)
+
+    spy.assert_not_called()
+
+
+def test_app_groups_records_by_depth_milestone(app_runner, mocker: MockerFixture):
+    """
+    It groups fetched records by depth milestone and calculates stats for each group
+    separately.
+    """
+
+    event = ScheduledDepthEvent(
+        asset_id=0,
+        company_id=0,
+        top_depth=0.0,
+        bottom_depth=0.0,
```

Comment:
The event for this test is arbitrary, because the api request is mocked. The only purpose is to test that `App groups fetched records by depth milestone and calculates stats for each group separately.`

Reply:
I've increased the precision of test values